cli/internal/runcache/runcache.go

```diff
@@ -190,6 +190,7 @@ func (tc TaskCache) ReplayLogFile(prefixedUI *cli.PrefixedUi, progressLogger hcl
 // This is called if the task exited with an non-zero error code.
 func (tc TaskCache) OnError(terminal *cli.PrefixedUi, logger hclog.Logger) {
 	if tc.taskOutputMode == util.ErrorTaskOutput {
+		terminal.Output(fmt.Sprintf("cache miss, executing %s", ui.Dim(tc.hash)))
```

Comment:
This is the existing implementation of `errors-only`. We don't know if the task has errored until it has run, so we buffer the log and replay in the event of an error

Reply:
I should add that it is a cache miss because `OnError` is only called after execution encountered an error. Given that we're executing, we know we didn't hit the cache

Reply:
Got it. I wish all the cache hit/miss/bypass messages were in `ec.exec()` rather than in the taskcache methods, it would make the flow a lot clearer